modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingProviderProducer.java

```diff
@@ -87,30 +92,34 @@ public int getSize() {
    * @return
    */
   private <R> R performWithConnection(Function<Object, R> function) {
-    ConnectionSupplier connectionSupplier = getConnectionSupplier();
+    connectionSupplier = getConnectionSupplier();
     Object connection = getConnection(connectionSupplier);
     try {
       R result = function.apply(connection);
       connectionSupplier.close();
       return result;
     } catch (Exception e) {
       if (isFirstPage) {
-        closeDelegate(connection);
+        silentlyCloseDelegate(connection);
       }
       extractConnectionException(e).ifPresent(ex -> connectionSupplier.invalidateConnection());
-      throw new MuleRuntimeException(createStaticMessage(COULD_NOT_EXECUTE), e);
+      throw e;
     }
   }
 
   /**
    * Closes the delegate
    */
   @Override
-  public void close() throws IOException {
-    ConnectionSupplier connectionSupplier = getConnectionSupplier();
-    closeDelegate(getConnection(connectionSupplier));
-    connectionSupplier.close();
-    connectionSupplierFactory.dispose();
+  public void close() {
+    try {
+      delegate.close(connectionSupplier.getConnection());
+    } catch (Exception e) {
+      throw new MuleRuntimeException(createStaticMessage(COULD_NOT_OBTAIN_A_CONNECTION), e);
+    } finally {
```

Comment:
log any potential failures in this block, and continue

Reply:
`public void close() throws IOException {
    ConnectionSupplier connectionSupplier = null;
    try {
      connectionSupplier = connectionSupplierFactory.getConnectionSupplier();
      delegate.close(connectionSupplier.getConnection());
    } catch (Exception e) {
      throw new MuleRuntimeException(createStaticMessage(COULD_NOT_OBTAIN_A_CONNECTION), e);
    } finally {
      if (connectionSupplier != null) {
        connectionSupplier.close();
      }
      connectionSupplierFactory.dispose();
    }
}`
That was the original code, before the other PR. Logging failures would be breaking backwards, right?

Reply:
ok, yes. You're right